OptimizelySDK/Event/EventFactory.cs

```diff
@@ -192,18 +192,40 @@ ProjectConfig config
             }
 
             var attributesList = new List<VisitorAttribute>();
+            var invalidUserAttributeKeys = new List<string>();
+            var notRegisteredAttributeKeys = new List<string>();
 
             if (userAttributes != null)
             {
-                foreach (var validUserAttribute in userAttributes.Where(attribute =>
-                             Validator.IsUserAttributeValid(attribute)))
+                foreach (var userAttribute in userAttributes)
                 {
-                    var attributeId = config.GetAttributeId(validUserAttribute.Key);
-                    if (!string.IsNullOrEmpty(attributeId))
+                    var attributeId = config.GetAttributeId(userAttribute.Key);
+                    if (string.IsNullOrEmpty(attributeId))
+                    {
+                        notRegisteredAttributeKeys.Add(userAttribute.Key);
+                        continue;
+                    }
+
+                    if (!Validator.IsUserAttributeValid(userAttribute))
                     {
-                        attributesList.Add(new VisitorAttribute(attributeId, validUserAttribute.Key,
-                            CUSTOM_ATTRIBUTE_FEATURE_TYPE, validUserAttribute.Value));
+                        invalidUserAttributeKeys.Add(userAttribute.Key);
+                        continue;
                     }
+
+                    attributesList.Add(new VisitorAttribute(attributeId, userAttribute.Key,
+                        CUSTOM_ATTRIBUTE_FEATURE_TYPE, userAttribute.Value));
+                }
+
+                if (notRegisteredAttributeKeys.Count > 0 && !(logger is null))
+                {
+                    logger.Log(LogLevel.WARN,
+                        $"User attributes: {string.Join(", ", notRegisteredAttributeKeys.ToArray())} are not supported by the datafile and will not be used.");
```

Comment:
Not sure if we want to log all these here. I see some clients have 100+ attributes not all registered to us. Each event will dump a bunch of these messages. Let's discuss.